get_block_details.py

```diff
@@ -0,0 +1,189 @@
+    # store it as the title
+    # else, move to the next block of code
+
+
+    # new variable declared as 'number'
+    # finding the last ':' in the 'title'
+    number = title.rfind(':')
+    # returns the title where ':' is found
+    return title[number + 1:]
+
+def shorten_rc_title(title):
+    """
+    Gets a PV title by shortening its address to the last segment.
+
+    Args:
+        title: The PV address as string.
+
+    Returns: The last segment of the input PV address as string.
+
+    """
+    # recieve the new value as a string for the RC value
+    # split string of the value at each colon
+    # look at each seperate string
+    # if there is a match on the block name
+    # new condition, if there is a match 'RC'
+    # new condition, if there is a value
+    # store it as the title
+    # else, move to the next block of code
+
+
+    # new variable declared as 'number'
+    # finding the last ':' in the 'title'
+
+    title = title.split(':')
+    rc_name = "RC"
+    value_name = ["HIGH.VAL", "LOW.VAL", "INRANGE.VAL", "ENABLED.VAL"]
+    title_list = []
+
+    if rc_name in title:
+        title_list.append(title[-3])
+        title_list.append(rc_name)
+        for name in value_name:
+            if name in title:
+                title_list.append(name)
+                title = ':'.join(title_list)
+                return title
+    else:
+        title = title[-1]
+        return title
+
+
+def get_rc_values_for_block(block_name, pvs):
+    """Search pvs for RC values for given block and return them"""
+    for k, v in pvs.items():
+        block_names = k.split(':')
+        if block_name == block_names[0]:
+            print block_name, v
+    return v
+
+def get_rc_values_for_block_from_object(block_name, pvs):
+    """Search pvs for RC values for given block and return them"""
+    values = []
+    items = pvs.items()
+    for k, v in items:
+        test_block_name = k.split(':')[0]
+        if block_name == test_block_name:
+            block_value = v.get_value()
+            values.append(block_value)
+    return values
+
+
+def set_rc_values_for_block_from_pvs(block_object, pvs):
+    """Search pvs for RC values for given block and return them"""
+    block_name = block_object.get_name()
+    items = pvs.items()
+
+    for k, v in items:
+        test_block_name = k.split(':')[0]
+        if block_name == test_block_name:
+            if "LOW.VAL" == k.split(':')[-1]:
+                low_rc = block_object.set_rc_low(v.get_value())
+            if "HIGH.VAL" == k.split(':')[-1]:
+                high_rc = block_object.set_rc_high(v.get_value())
+            if "INRANGE.VAL" == k.split(':')[-1]:
+                inrange_rc = block_object.set_rc_inrange(v.get_value())
+                return low_rc, high_rc, inrange_rc
+
+def set_rc_values_for_blocks(block_objects, pvs):
+    """Set all RC values for all the given blocks"""
+    for object in block_objects:
+        set_rc_values_for_block_from_pvs(object,pvs)
```

Comment:
`object` is a type in python. We should use a different name to prevent shadowing,